packages/core/src/components/menu/menuItem.tsx

```diff
@@ -77,6 +77,28 @@ export interface IMenuItemProps extends ActionProps, LinkProps {
      */
     labelElement?: React.ReactNode;
 
+    /**
+     * Changes the ARIA `role` property structure of this MenuItem to accomodate for various
+     * different `role`s of the parent Menu `ul` element.
+     *
+     * If `menuitem`, role structure becomes:
+     *
+     * `<li role="none"`
+     *     `<a role="menuitem"`
+     *
+     * which is proper role structure for a `<ul role="menu"` parent (this is the default `role` of a `Menu`).
+     *
+     * If `listoption`, role structure becomes:
+     *
+     * `<li role="option"`
+     *     `<a role=undefined`
+     *
+     *  which is proper role structure for a `<ul role="listbox"` parent, or a `<select>` parent.
+     *
+     * @default "menuitem"
+     */
+    roleConfig?: "menuitem" | "listoption";
```

Comment:
prefer to avoid names like "config" in prop / variable names, as this is all essentially "configuration". instead we use "kind" in other places in Blueprint, and I think it could work well here:

```suggestion
    roleKind?: "menuitem" | "listoption";
```

Reply:
what are your thoughts about `roleStructure`? or do you still prefer `roleKind`?

Reply:
Alright, I don't have a strong opinion between the two options, let's go with what you have now 👍🏽